libs/sarif/src/schema/reporting_descriptor.rs

```diff
@@ -0,0 +1,430 @@
+    ///
+    /// assert_eq!(
+    ///     descriptor.message_strings.get("default").unwrap().text,
+    ///     "Expected type did not match the received type."
+    /// );
+    /// ```
+    #[must_use]
+    pub fn with_message_string(
+        mut self,
+        key: impl Into<Cow<'s, str>>,
+        message_string: MultiformatMessageString<'s>,
+    ) -> Self {
+        self.message_strings.insert(key.into(), message_string);
+        self
+    }
+
+    /// Sets the help of the `ReportingDescriptor`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use sarif::schema::{MultiformatMessageString, ReportingDescriptor};
+    ///
+    /// const ERROR_HELP_MESSAGE: &str =
+    ///     "This error occurs when an expression was used in a place where the compiler expected an \
+    ///      expression of a different type. It can occur in several cases, the most common being \
+    ///      when calling a function and passing an argument which has a different type than the \
+    ///      matching type in the function declaration.";
+    ///
+    /// let descriptor = ReportingDescriptor::new("E0308")
+    ///     .with_help(MultiformatMessageString::new(ERROR_HELP_MESSAGE));
+    ///
+    /// assert_eq!(descriptor.help.unwrap().text, ERROR_HELP_MESSAGE);
+    /// ```
+    #[must_use]
+    #[expect(
+        clippy::missing_const_for_fn,
+        reason = "destructor cannot be evaluated at compile-time"
+    )]
+    pub fn with_help(mut self, help: MultiformatMessageString<'s>) -> Self {
+        self.help = Some(help);
+        self
+    }
+
+    /// Sets the help URI of the `ReportingDescriptor`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use sarif::schema::ReportingDescriptor;
+    /// use url::Url;
+    ///
+    /// let descriptor = ReportingDescriptor::new("E0308").with_help_uri(Url::parse(
+    ///     "https://doc.rust-lang.org/error-index.html#E0308",
+    /// )?);
+    ///
+    /// assert_eq!(
+    ///     descriptor.help_uri.unwrap().as_str(),
+    ///     "https://doc.rust-lang.org/error-index.html#E0308"
+    /// );
+    /// # Ok::<(), url::ParseError>(())
+    /// ```
+    #[must_use]
+    pub fn with_help_uri(mut self, help_uri: impl Into<Url>) -> Self {
+        self.help_uri = Some(help_uri.into());
+        self
+    }
+}
+
+#[cfg(all(feature = "serde", test))]
+mod tests {
+    use alloc::{
+        format,
+        string::{String, ToString},
+    };
+    use std::{io, process::Command};
+
+    use url::Url;
+
+    use crate::schema::{MultiformatMessageString, ReportingDescriptor};
+
+    fn error_code_to_uri(code: u32) -> String {
+        format!("https://doc.rust-lang.org/error-index.html#E{code:04}")
+    }
+
+    fn error_code_to_help(code: u32) -> io::Result<String> {
+        let explain_command = Command::new("rustc")
+            .arg("--explain")
+            .arg(format!("E{code:04}"))
+            .output()?;
+
+        Ok(String::from_utf8_lossy(&explain_command.stdout).to_string())
+    }
```

Comment:
I like this! :D

Reply:
Sadly, this is nothing we can use at runtime as we don't necessarily have `rustc` in the environment, but the information should be there in the JSON output of the compiler, though.